### DISCUSSION TITLE
code-server with Remote Development using SSH

### DISCUSSION THREAD

**Discussion by coatmaker618** [coder/code-server#2838] [Q&A]
I'm in the process of setting this up on a self-hosted (local) server so I can access it when I'm not on the local network (and when other options like VPN//SSH aren't options).  However when I AM on the local network, it'd be nice to be able to use VSCode instead of a web browser.

Is it possible to use "Remote Development using SSH" (seen here: https://code.visualstudio.com/docs/remote/ssh)  to connect a local VSCode to the installed code-server?  I notice the non nginx approach is SSH, but I'm unclear if it "just works" with vscode remote dev, or if there's other things I hafta do.

Thanks & so glad to have this :)

**Comment by code-asher** [comment]
The remote extensions aren't designed to connect to code-server but you
can still use them. When they connect they'll set up their own server
separate from code-server.

**Reply by coatmaker618** [reply]
Hrm, ok.  So it will create a separate instance.  So it sounds like if I wanna use both, I just have to deal with two separate environments, gotcha.  Thanks for the heads up!

**Reply by code-asher** [reply]
Ah yup that's the main downside. We do have an issue for developing our own remote extensions that use code-server as the backend (https://github.com/cdr/code-server/issues/1315) but it's not on the roadmap.

You could try sharing settings/extensions between them but I'm not sure if that would play well.

**Reply by coatmaker618** [reply]
@code-asher good to know that they're unlikely to play well so I shouldn't go in with high expectations!

While there may be no roadmap, a quick read seems to indicate that you've made a lot of progress on this in the last year and that the current thought is that it would be easier to make this the target of a VSCode client trying to connect rather than have it connect with a VSCode server?

**Reply by code-asher** [reply]
To be honest we haven't made any progress at all on the remote extensions despite all the comments on that issue. :stuck_out_tongue: 

But yeah, the idea is that we'd make our own VS Code remote extensions that mimic exactly what the official remote extensions do except they would use code-server as a backend instead of their own server. So you'd be using a regular VS Code client connecting to a code-server instance.

The reason it'll be so much work is that the official remote extensions are closed source so we'd have to implement our own from scratch. :cry:
